lib/utils/session_keep_alive.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "services"
+require "cluster"
+
+Services.mongo!
+
+module Utils
+  # Lets you wrap a session around a potentially long running query,
+  # and that session is kept alive by a refresh-loop in a separate thread.
+  # Intended usage:
+  # require "utils.session_keep_alive"
+  # ska = Utils::SessionKeepAlive.new(60)
+  # ska.run do
+  #   # Mongo operation at risk of timing out:
+  #   Cluster.where(long_q).no_timeout.each do |cluster|
+  #     {...}
+  #   end
+  # end
+  class SessionKeepAlive
```

Comment:
Worth adding a comment about how we tried to test that this fixes the disappearing-cursor issue but failed to successfully test that.